### PR TITLE
Move tabs over to new style

### PR DESCRIPTION
**Pull request by colinlohner**
**Overview**
Update tabs to use new theming style

**Screenshots (if applicable)**
V6 Theme
image

V5 theme
image

**Documentation**

- [ ] Updated Typescript types and/or component PropTypes
- [ ] Added / modified component docs
- [ ] Added / modified Storybook stories

### CODE REVIEW

File: src/themes/classic/components/tab.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  },
+
+  _focus: {
+    boxShadow: 'shadows.focusRing'
+  }
+}
+
+const appearances = {
+  primary: tabStyles,
+  secondary: tabStyles,
```
Comment (akleiner2) [inline src/themes/classic/components/tab.js:31]: Does the default theme have a `secondary` appearance for tabs if they're the same?

File: src/themes/classic/components/tab.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+export default {
+  baseStyle,
+  appearances
```
Comment (akleiner2) [inline src/themes/classic/components/tab.js:36]: For the sake of consistency elsewhere, should we have a `const sizes = {}`?

File: src/themes/classic/components/tab.js (hunk 1 above)
Comment (colinlohner) [reply src/themes/classic/components/tab.js:31]: The default theme has two yes

File: src/tabs/src/Tab.js [hunk 3]
```diff
@@ -1,27 +1,60 @@
...
+import useStyleConfig from '../../hooks/use-style-config'
 import safeInvoke from '../../lib/safe-invoke'
 import warning from '../../lib/warning'
-import useTabApperance from '../../theme/src/hooks/useTabApperance'
 import { Text } from '../../typography'
 
 const noop = () => {}
 
+const getInternalStyles = (direction) => ({
+  fontWeight: 500,
```
Comment (mshwery) [inline src/tabs/src/Tab.js:11]: Thinking `fontWeight` should probably be in `baseStyle` in the theme?

File: src/tabs/src/Tab.js [hunk 4]
```diff
@@ -73,12 +106,14 @@ const Tab = memo(
 
     return (
       <Text
-        className={tabClassName}
+        className={themedClassName}
```
Comment (mshwery) [inline src/tabs/src/Tab.js:109]: Can we do `cx(className, themedClassName)` to make sure we support external classnames being passed in?

File: src/themes/classic/components/tab.js (hunk 1 above)
Comment (mshwery) [reply src/themes/classic/components/tab.js:31]: Seems like if there is no difference in appearances for v6 theme, I would move this all into `baseStyle` instead

File: src/themes/default/components/tab.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    paddingBottom: '16px',
+    paddingLeft: '2px',
+    paddingRight: '2px',
+    position: 'relative',
+
+    ':not(:last-child)': {
+      marginRight: '20px'
+    },
+    
+    ':before': {
```
Comment (mshwery) [inline src/themes/default/components/tab.js:18]: Can we just bridge support for this by adding pseudo placeholders for `_before` and `_after`?

File: src/themes/classic/components/tab.js (hunk 2 above)
Comment (colinlohner) [reply src/themes/classic/components/tab.js:36]: hmm not sure how i feel about having unused objects floating around

**Review by mshwery — APPROVED** [review]
Lookin good! Left a few minor comments.

**Review by akleiner2 — APPROVED** [review]

**Pull request merged**